src/logger/logger.service.js

```diff
@@ -8,46 +8,77 @@
  * Service in the vega-lite-ui.
  */
 angular.module('vlui')
-  .service('Logger', function ($location, $window, consts) {
+  .service('Logger', function ($location, $window, consts, Analytics) {
 
     var service = {};
 
+    service.levels = {
+      OFF: {id:'OFF', rank:0},
+      TRACE: {id:'TRACE', rank:1},
+      DEBUG: {id:'DEBUG', rank:2},
+      INFO: {id:'INFO', rank:3},
+      WARN: {id:'WARN', rank:4},
+      ERROR: {id:'ERROR', rank:5},
+      FATAL: {id:'FATAL', rank:6}
+    };
+
     service.actions = {
-      INITIALIZE: 'INITIALIZE',
-      UNDO: 'UNDO',
-      REDO: 'REDO',
-      DATASET_CHANGE: 'DATASET_CHANGE',
-      CHART_MOUSEOVER: 'CHART_MOUSEOVER',
-      CHART_MOUSEOUT: 'CHART_MOUSEOUT',
-      CHART_RENDER: 'CHART_RENDER',
-      CHART_EXPOSE: 'CHART_EXPOSE',
-      CHART_TOOLTIP: 'CHART_TOOLTIP',
-      CHART_TOOLTIP_END: 'CHART_TOOLTIP_END',
-      BOOKMARK_ADD: 'BOOKMARK_ADD',
-      BOOKMARK_REMOVE: 'BOOKMARK_REMOVE',
-      BOOKMARKS_CLEAR: 'BOOKMARKS_CLEAR',
-
-      NULL_FILTER_TOGGLE: 'NULL_FILTER_TOGGLE',
-      TRANSPOSE_TOGGLE: 'TRANSPOSE_TOGGLE',
-      SORT_TOGGLE: 'SORT_TOGGLE',
-      MARKTYPE_TOGGLE: 'MARKTYPE_TOGGLE',
-      LOG_TOGGLE: 'LOG_TOGGLE',
-
-      FUNC_CHANGE: 'FUNC_CHANGE',
-      // Polestar only
-      SPEC_CHANGE: 'SPEC_CHANGE',
-      FIELD_DROP: 'FIELD_DROP',
-      MARK_CHANGE: 'MARK_CHANGE',
-      // Voyager only
-      FIELDS_CHANGE: 'FIELDS_CHANGE',
-      FIELDS_RESET: 'FIELDS_RESET',
-      DRILL_DOWN_OPEN: 'DRILL_DOWN_OPEN',
-      DRILL_DOWN_CLOSE: 'DRILL_DOWN_CLOSE',
-      CLUSTER_SELECT: 'CLUSTER_SELECT',
-      LOAD_MORE: 'LOAD_MORE'
+      // DATA
+      INITIALIZE: {cat: 'DATA', id: 'INITIALIZE', level: service.levels.DEBUG},
```

Comment:
+1 for `category`, we use `cat` in the WordPress REST API and it trips me up every time :)